lib/upnp/upnp.go

```diff
@@ -107,7 +107,17 @@ func Discover(ctx context.Context, _, timeout time.Duration) []nat.Device {
 		for _, deviceType := range []string{"urn:schemas-upnp-org:device:InternetGatewayDevice:1", "urn:schemas-upnp-org:device:InternetGatewayDevice:2"} {
 			wg.Add(1)
 			go func(intf net.Interface, deviceType string) {
-				discover(ctx, &intf, deviceType, timeout, resultChan)
+				// For each protocol, try to discover IPv6 gateways.
+				discover(ctx, &intf, deviceType, timeout, resultChan, true)
+				wg.Done()
+			}(intf, deviceType)
+		}
+
+		for _, deviceType := range []string{"urn:schemas-upnp-org:device:InternetGatewayDevice:1", "urn:schemas-upnp-org:device:InternetGatewayDevice:2"} {
```

Comment:
I don't see any difference in the device type strings , so you could just kick off two goroutines in a single loop body?

Reply:
You're right, that works. Done